### ISSUE TITLE
New Waveshare (b) devices use Different CH422G outputs 

### ISSUE THREAD

**Issue by bondango** [esp-arduino-libs/ESP32_IO_Expander#5]
The new Waveshare revision (B) devices for example esp 4.3(b) use the 8 directional i/o's and also the General purpose Output Pins of the CH422G. 
As per CH422G Datasheet :-
The pins OC3-OC0 of CH422 are push-pull or open-drain output pins, and are push-pull output pins by
default.

Waveshare schematic :-
![1728054451643](https://github.com/user-attachments/assets/75ce7aa4-b31a-4bd7-bfce-ab74295aac40)

Datasheet Layout:-
![1728054329375](https://github.com/user-attachments/assets/ba99ed1c-6d8e-4785-a47a-84666cf990bb)

On the new revision boards OC0 and OC1 are used as digital Outputs to drive BC817-40's  via PC817X3's

The current library has support for the 8 Directional I/Os but not the OCx Outputs.

Marty

**Comment by Lzw655** [comment]
Hi @bondango, sorry for the late. I tried modifying the driver according to the CH422G manual so that it can control the OCx (pin number: 8-11) output level. However, since I don't have the hardware, could you please help me test it? Thank you very much!

**Comment by bondango** [comment]
> Thank

Yes I will try and report hopefully this weekend

**Comment by bondango** [comment]
I did try a simple 1 second pause loop

expander - digitalWrite(8, HIGH);
expander - digitalWrite(9, HIGH);
delay(500);
expander - digitalWrite(8, LOW);
expander - digitalWrite(9, LOW);
delay(500);

simple light test
power gnd >>>output GND
+12v >>> bulb
blub >>> DI0

But nothing on the DO0 and D01 outputs
inputs are working fine

**Comment by AdVirMachina** [comment]
I have installed this and am using it against a Waveshare s3-4.3LCD board. I created a simple loop which turns off the backlight, waits 1 second, and turns it back on, waits 1 second and then loops. This particular board is using the CH422G to also manage which USB port is active (the board sports two). I am receiving USB disconnects when monitoring the application as it transitions the backlight from off to on. I suspect this has to do with the code assuming the initial state of the pins and not reading it into the "cache". I'm trying to instrument the code to validate this.

TBH, I don't understand where the register values have come from. The PDFs of the datasheet I've found don't mention them. In fact, they don't really outline an I2C interface but rather some unusual two wire interactions. Can someone help me source the I2C interface information? I can't find the example program cited in the datasheet either.
